Scripts/Process/ProcessContainer.cs

```diff
@@ -0,0 +1,51 @@
+﻿namespace VRTK.Core.Process
+{
+    using UnityEngine;
+
+    /// <summary>
+    /// The ProcessContainer acts as a container for a process that can be utilized within a Unity Inspector.
+    /// </summary>
+    public abstract class ProcessContainer
+    {
+        [SerializeField]
+        protected Object process;
+    }
+
+    /// <summary>
+    /// The ProcessContainer acts as a container for a given interface type.
+    /// </summary>
+    /// <typeparam name="TInterface">The type of container for the interface.</typeparam>
+    public abstract class ProcessContainer<TInterface> : ProcessContainer
+    {
+        public TInterface Interface
+        {
+            get
+            {
+                return process == null ? _interface : (TInterface)(object)process;
+            }
+            set
+            {
+                Object @object = value as Object;
+                if (@object == null)
+                {
+                    _interface = value;
+                    process = null;
+                }
+                else
+                {
+                    process = @object;
+                    _interface = default(TInterface);
+                }
+            }
+        }
+        private TInterface _interface;
+    }
+
+    /// <summary>
+    /// The ProcessInterfaceContainer is the proxy class used to make a process interface appear in the Unity Inspector.
+    /// </summary>
+    [System.Serializable]
+    public sealed class ProcessInterfaceContainer : ProcessContainer<IProcessable>

```

Comment:
Once we do the rename to `InterfaceContainer` we should move this into its own file.

Reply:
👍